src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -390,12 +266,21 @@ public Task<Compilation> GetCompilationAsync(CancellationToken cancellationToken
         }
 
         /// <summary>
-        /// Determines if the compilation returned by <see cref="GetCompilationAsync"/> has all the references it's expected to have.
+        /// Determines if the compilation returned by <see cref="GetCompilationAsync"/> has full information.
+        /// see <see cref="ProjectInfo.IsComplete"/> for more info.
+        /// </summary>
+        public Task<bool> IsCompleteAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return _solution.IsCompleteAsync(this, cancellationToken);
+        }
+
+        /// <summary>
+        /// Determines if the compilation returned by <see cref="GetCompilationAsync"/> and all its referenced projects have full information.
+        /// see <see cref="IsCompleteAsync(CancellationToken)"/> and <see cref="ProjectInfo.IsComplete"/> for more info.
         /// </summary>
-        // TODO: make this public
-        internal Task<bool> HasCompleteReferencesAsync(CancellationToken cancellationToken = default(CancellationToken))
+        public Task<bool> IsDependentCompleteAsync(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
we already have IsCompleteAsync. I am open to other name if you don't like. I was following pattern others use.